eng/SnippetGenerator/Program.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SnippetGenerator
+{
+    public class Program
+    {
+        [Option(ShortName = "u")]
+        public string Markdown { get; set; }
+
+        [Option(ShortName = "s")]
+        public string Snippets { get; set; }
+
+        public static int Main(string[] args)
+        {
+            return CommandLineApplication.Execute<Program>(args);
+        }
+
+        public async Task OnExecuteAsync()
+        {
+            Console.WriteLine($"Processing {Markdown}");
+
+            var text = File.ReadAllText(Markdown);
+            var snippets = await GetSnippetsInDirectory(Snippets);
+            Console.WriteLine($"Discovered snippets:");
```

Comment:
I wonder if we should use a convention to denote that a given region is intended as a snippet.    I don't have a good suggestion here, as I acknowledge that this could quickly devolve into a lot of noise.

Reply:
Currently there are no 'region's in samples so I didn't worry about that

Reply:
I had the same feedback. I suggest we use `Snippet:` for regions that will go into readme files.

Reply:
Fair, but its not clear why we're using regions, unless you understand the convention.   Without that context, they'll just see a bunch of fenced in areas of code that I would assume are unimportant and meant to be collapsed when reading the sample.

Reply:
I'm not sure what you mean. They'll see regions anyway, right?

Reply:
I mean that, as a general rule, when you read code, things that are unimportant and boilerplate are typically what is hidden away in regions.   The first thing that I do is typically squish region blocks so that I can concentrate on the "important stuff".    I could be unique in that, but I tend to believe that is a mainstream behavior from what I've experienced in the wild.

Without knowledge that our regions are surrounding code important enough to pull out as snippets, I fear that using them in samples would be confusing and frustrating to developers trying to read and understand them.

Reply:
There are no existing regions in our samples so I didn't bother.

Reply:
> I'm not sure what you mean. They'll see regions anyway, right?

I think I understand why you were asking.   The comment that I left following that was in response to Pavel.   Your comment snuck in there while I was typing.   Apologies for the confusion.

Reply:
I'm fine adding the prefix.